drivers/wifi/infineon/Kconfig.airoc

```diff
@@ -17,6 +16,24 @@ menuconfig WIFI_AIROC
 
 if WIFI_AIROC
 
+choice
```

Comment:
Can this configuration replace with `DT_ON_BUS`?

Reply:
yes looks like it a good idea to select spi/sdio(sdhc) by DT_ON_BUS, so it will simplify bus configuration. For now we need disable one, enable another...:
```
CONFIG_AIROC_WIFI_BUS_SDIO=n
CONFIG_AIROC_WIFI_BUS_SPI=y
```

Reply:
@soburi - Looks like most of the cases can be replaced with DT_ON_BUS().  The two I don't know how to address are in CMakeLists.txt, where the appropriate bus interface source file is selected, and making the option AIROC_WIFI_BUS_SPI_DATA_IRQ_MUX available in Kconfig.airoc.  The latter enables logic to change use of the GPIO line in the special case where the SPI data and IRQ are shared, which is the case for the Pico W.
Do you have any suggestions for how to handle those?

Reply:
Can it be configured by pinctrl?
IMO, The pin function should not be configured in the wifi driver.

Reply:
> yes looks like it a good idea to select spi/sdio(sdhc) by DT_ON_BUS, so it will simplify bus configuration. For now we need disable one, enable another...:
> 
> ```
> CONFIG_AIROC_WIFI_BUS_SDIO=n
> CONFIG_AIROC_WIFI_BUS_SPI=y
> ```

@npal-cy - Yes, that's why I made it a choice.  Now that I know about DT_ON_BUS, yes, that makes things cleaner ... if we can figure out how to configure cmake and the data/IRQ multiplexing.

Reply:
> Can it be configured by pinctrl?
> IMO, The pin function should not be configured in the wifi driver.

I struggled with this issue.  On the Pico W, GPIO 24 is used for bidirectional data as well as HOST_WAKE.  In order to make this work, the pin has to be assigned to SIO (function 5) when serving as an interrupt, but switched to the PIO (function 6 or 7, depending) when used for SPI data.  The Pico SPI driver doesn't know about this silliness;  it seems to belong in the layer within the WiFi driver that deals with the SPI bus.  This was the best compromise I could come up with.  I'd be happy to hear any cleaner solutions.

Reply:
> @npal-cy - Yes, that's why I made it a choice. Now that I know about DT_ON_BUS, yes, that makes things cleaner ... if we can figure out how to configure cmake and the data/IRQ multiplexing.

I means to keep AIROC_WIFI_BUS_SPI/AIROC_WIFI_BUS_SDIO, make them hided and auto enabled by on_bus, something like snipped below (not tested...)
```
config AIROC_WIFI_BUS_SDIO
    bool
    default $(dt_compat_on_bus,$(DT_COMPAT_INFINEON_AIROC_WIFI),sd)
    select SDIO_STACK
	select SDHC

config AIROC_WIFI_BUS_SPI
    bool
    default $(dt_compat_on_bus,$(DT_COMPAT_INFINEON_AIROC_WIFI),spi)
    select SPI

```

Reply:
@ThreeEights 

> Do you have any suggestions for how to handle those?

We can use `pinctrl_apply_state` with a private state.

https://github.com/zephyrproject-rtos/zephyr/blob/2f47de60a0a7038eed1bfe11ced16ae51444434e/drivers/sdhc/imx_usdhc.c#L176

Reply:
@soburi - I think I've come up with a better solution.  I've changed most of what had been Kconfig settings to rely instead on devicetree.  See what you think of this version - changes in:
```
dts/bindings/wifi/infineon,airoc-wifi-spi.yaml
boards/raspberrypi/rpi_pico/rpi_pico_rp2040_w_defconfig
boards/raspberrypi/rpi_pico/rpi_pico_rp2040_w.dts
drivers/wifi/infineon/Kconfig.airoc
drivers/wifi/infineon/airoc_whd_hal_spi.c
drivers/wifi/infineon/airoc_wifi.h
```

Reply:
@ThreeEights 

The overall bus structure has been improved by organizing it.
However, this pinctrl topic is separate from this and is the last piece of the puzzle.
It should be applied to avoid leaving the rp2040-dependent code here.

Reply:
@soburi @npal-cy - I reworked control of the GPIO shared between data and the HOST_WAKE interrupt to use pinctrl instead of direct control.  Since that is specific to the SPI code in the Infineon driver, the pinctrl configurations must be handled at that level, which makes the RPI Pico dts look a bit peculiar.  Those changes affect:
```
boards/raspberrypi/rpi_pico/rpi_pico_rp2040_w.dts
drivers/wifi/infineon/airoc_whd_hal_spi.c
drivers/wifi/infineon/airoc_wifi.c
drivers/wifi/infineon/airoc_wifi.h
```
I also modified drivers/wifi/infineon/Kconfig.airoc so the bus selection now depends only on the dt_compat_on_bus extension.